src/model.cpp

```diff
@@ -49,6 +70,25 @@ using LibMatrix::uvec3;
     } \
 } while(0);
 
+const int isBigEnd=1;
+#define is_bigendian() ((*(char*)&isBigEnd) == 0)
+
+float FloatSwap(float f)
+{
+    union
+    {
+        float f;
+        unsigned char b[4];
+    } dat1, dat2;
+
+    dat1.f = f;
+    dat2.b[0] = dat1.b[3];
```

Comment:
It turns out that, per the C++ standard, it's undefined behavior to read from a union member that wasn't explicitly written to before (although most compilers seem to deal with it OK). We can safely cast a float to char*, though, and achieve the same result:

```
float FloatSwap(float f)
{
    float ret;
    char *fbytes = reinterpret_cast<char*>(&f);
    char *retbytes = reinterpret_cast<char*>(&ret);
   ...
}